options.py

```diff
@@ -81,6 +89,9 @@ class ValueOption[J: JSON](BaseOption):
     def __init__(self) -> None:
         raise NotImplementedError
 
+    def _to_json(self) -> JSON:
+        return cast(JSON, self.value)
```

Comment:
is this cast actually necessary? `self.value` should be json already, surely you can widen it.

also as a nitpick, a cast has a performance impact, it does an actual function call. if you don't need the result straight away (e.g. cause you're returning it), just a type ignore is slightly better